### PR TITLE
 [IMP] datepicker issue fixed for khmer langauge

### PR DESCRIPTION
**Pull request by sja-odoo**
Issue: https://www.odoo.com/web#id=747140&view_type=form&model=project.issue&action=609&menu_id=4720

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/web/static/lib/moment/locale/km.js [hunk 1]
```diff
@@ -15,7 +15,7 @@ var km = moment.defineLocale('km', {
     monthsShort: 'មករា_កុម្ភៈ_មីនា_មេសា_ឧសភា_មិថុនា_កក្កដា_សីហា_កញ្ញា_តុលា_វិច្ឆិកា_ធ្នូ'.split('_'),
     weekdays: 'អាទិត្យ_ច័ន្ទ_អង្គារ_ពុធ_ព្រហស្បតិ៍_សុក្រ_សៅរ៍'.split('_'),
     weekdaysShort: 'អាទិត្យ_ច័ន្ទ_អង្គារ_ពុធ_ព្រហស្បតិ៍_សុក្រ_សៅរ៍'.split('_'),
-    weekdaysMin: 'អាទិត្យ_ច័ន្ទ_អង្គារ_ពុធ_ព្រហស្បតិ៍_សុក្រ_សៅរ៍'.split('_'),
+    weekdaysMin: 'អា_ច_អ_ពុ_ព្រ_សុ_ស'.split('_'),
```
Comment (rim-odoo) [inline addons/web/static/lib/moment/locale/km.js]: Shouldn't this be fixed in the upstream library: https://github.com/moment/moment/blob/develop/locale/km.js#L18 ?

Comment (mart-e) [reply addons/web/static/lib/moment/locale/km.js]: Yes, exactly, please report the changes upstream and, if accepted, we will update our locales

File: odoo/addons/base/data/res.lang.csv [hunk 2]
```diff
@@ -35,7 +35,7 @@ base.lang_de_CH,"German (CH) / Deutsch (CH)",de_CH,de_CH,"Left-to-Right","[3,3]"
 "base.lang_it","Italian / Italiano","it_IT","it","Left-to-Right","[3,0]",",",".","%d/%m/%Y","%H:%M:%S","True"
 "base.lang_ja_JP","Japanese / 日本語","ja_JP","ja","Left-to-Right","[3,0]",".",",","%Y年%m月%d日","%H時%M分%S秒","True"
 "base.lang_kab_DZ","Kabyle / Taqbaylit","kab_DZ","kab","Left-to-Right","[]",".",",","%m/%d/%Y","%I:%M:%S %p","True"
-"base.lang_km","Khmer / ភាសាខ្មែរ","km_KM","km","Left-to-Right","[3,0]",".",",","%d %B %Y","%H:%M:%S","True"
+"base.lang_km","Khmer / ភាសាខ្មែរ","km_KH","km","Left-to-Right","[3,0]",".",",","%d %B %Y","%H:%M:%S","True"
```
Comment (rim-odoo) [inline odoo/addons/base/data/res.lang.csv]: Seems legit
```bash
$ cat /etc/locale.gen | grep km
#km_KH UTF-8
```

**Comment by rim-odoo** [comment]
Partially merged here https://github.com/odoo/odoo/commit/3830621075805ff71bffe45f428b7318ab2769d2

**Comment by pedrobaeza** [comment]
This makes all the DBs that makes `-u all` to fail due to this restriction: https://github.com/odoo/odoo/blob/e89fbb5c389d53deacb35911a783ce216601e44d/odoo/addons/base/res/res_lang.py#L218

**Comment by gustavovalverde** [comment]
Confirming issue reported by @pedrobaeza

**Comment by pedrobaeza** [comment]
A temporary solution is to perform following SQL on the DB:
```sql
UPDATE res_lang SET code='km_KH' WHERE code='km_KM';
```

**Comment by Yajo** [comment]
I wonder... if language code cannot be modified, why has it been modified? 🤔 Either the restriction is unnecessary, or this patch should not have landed in the first place.

**Comment by rim-odoo** [comment]
Revertedhttps://github.com/odoo/odoo/commit/2b32f52b48c58f237db1beeeb2012a662fdc3ce6, thanks for your reports

**Comment by pedrobaeza** [comment]
Thank you for the quick reaction. @hbrunn @gurneyalex can we force a sync on OCB for not waiting till tonight to fix the problem?

**Comment by rim-odoo** [comment]
Finally fixed in master  https://github.com/odoo/odoo/commit/40b879b75dfe271558e8ea05a3911b5be1ee80d6

File: addons/web/static/lib/moment/locale/km.js (hunk 1 above)
Comment (sja-odoo) [reply addons/web/static/lib/moment/locale/km.js]: https://github.com/moment/moment/issues/4353

**Pull request closed without merging**